openvdb/points/PointDataGrid.h

```diff
@@ -1163,73 +1163,103 @@ PointDataLeafNode<T, Log2Dim>::readBuffers(std::istream& is, bool fromHalf)
     this->readBuffers(is, CoordBBox::inf(), fromHalf);
 }
 
+
+////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////
+
+
+namespace page_stream
+{
+
+template <typename StreamT>
+inline void destroyPagedStream(const MetaMap& auxData, const Index index)
+{
+    // if paged stream exists, delete it
+    const std::string key("paged:" + std::to_string(index));
+    (const_cast<MetaMap&>(auxData)).removeMeta(key);
+}
+
+template <>
+inline void destroyPagedStream<compression::PagedOutputStream>
+    (const MetaMap& auxData, const Index index)
+{
+    using TypedMetaT = TypedMetadata<compression::PagedOutputStream::Ptr>;
+
+    // if paged stream exists, flush and delete it
+    const std::string key("paged:" + std::to_string(index));
+    const TypedMetaT::ConstPtr it = auxData.getMetadata<TypedMetaT>(key);
+    if (it) {
+        it->value()->flush();
+        (const_cast<MetaMap&>(auxData)).removeMeta(key);
+    }
+}
+
+template <typename StreamT>
+inline StreamT&
+getOrInsertPagedStream(const MetaMap& auxData,
+                       const Index index)
+{
+    using TypedMetaT = TypedMetadata<typename StreamT::Ptr>;
+
+    const std::string key("paged:" + std::to_string(index));
+    typename TypedMetaT::ConstPtr it = auxData.getMetadata<TypedMetaT>(key);
+    if (it) {
+        return *(it->value());
+    }
+    else {
+        typename StreamT::Ptr pagedStream = std::make_shared<StreamT>();
+        (const_cast<MetaMap&>(auxData)).insertMeta(key, TypedMetaT(pagedStream));
+        return *pagedStream;
+    }
+}
+
+} // namespace page_stream
```

Comment:
@danrbailey be good for you to take a look through this - I started splitting out all the descriptor methods from the `Local` structs into a shared namespace too, but there's ever so slightly different implementations of some. Would be good to consolidate them